### PR TITLE
Changelog for release v2.0.0

### PR DESCRIPTION
**Pull request by emmenko**

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,3 +1,124 @@
...
+```
+
+...or pass a mapping function as the first argument to return custom shape of the injected props
+
+```js
+withApplicationState(applicationState => ({
+  projectKey: applicationState.project && applicationState.project.key,
+  userEmail: applicationState.user && applicationState.user.email,
+}))(MyComponent);
+```
```
Comment (dferber90) [inline CHANGELOG.md]: This documentation is nice! I'd move it to a more permanent place like a dedicated `.md` file and link to it from the  `README`. Then I would remove some details form the change log and also link to them from here.

I'm currently writing some documentation about `test-utils` myself. Separate docs for `ApplicationState` would be nice. I could also link to them from `test-utils` then 👍

File: CHANGELOG.md [hunk 2]
```diff
@@ -1,3 +1,124 @@
...
+// Before
+import AsyncLocaleData from '@commercetools-frontend/i18n/async-locale-data';
+
+// After
+import { AsyncLocaleData } from '@commercetools-frontend/i18n';
+```
+
+> This includes also the `<ApplicationShell>`, so now it's `import { ApplicationShell } from '@commercetools-frontend/application-shell';`
+
+### Published packages are bundled to ESM or CJS
```
Comment (dferber90) [inline CHANGELOG.md]: ~or~ *and

File: CHANGELOG.md (hunk 1 above)
Comment (emmenko) [reply CHANGELOG.md]: Feel free to do those changes 😉

File: CHANGELOG.md [hunk 3]
```diff
@@ -1,3 +1,124 @@
...
+  ),
+  // ...plus all other `@commercetools-frontend` packages
+];
+
+createWebpackConfigForDevelopment({
+  distPath,
+  entryPoint,
+  sourceFolders,
+});
+```
```
Comment (dferber90) [inline CHANGELOG.md]: I wouldn't include the instructions for how it was done previously as it's a bit confusing. I don't really understand what to do with this code-block. Delete it from my setup? Some parts of it are still required though, right?

I'd maybe do

```diff
const path = require('path');
const createWebpackConfigForDevelopment = require('@commercetools-frontend/mc-scripts/config/create-webpack-config-for-development');
const distPath = path.resolve(__dirname, 'dist');
const entryPoint = path.resolve(__dirname, 'src/index.js');

const sourceFolders = [
  path.resolve(__dirname, 'src'),
-  path.resolve(
-    __dirname,
-    'node_modules/@commercetools-frontend/application-shell'
-  ),
  // ...plus all other `@commercetools-frontend` packages
];

createWebpackConfigForDevelopment({
  distPath,
  entryPoint,
  sourceFolders,
});
```

File: CHANGELOG.md (hunk 1 above)
Comment (dferber90) [reply CHANGELOG.md]: ;)

Comment (tdeekens) [reply CHANGELOG.md]: ...but he already have this documentation here or what part are we missing? https://github.com/commercetools/merchant-center-application-kit/tree/master/packages/application-shell-connectors/src/components/application-state

Comment (dferber90) [reply CHANGELOG.md]: Nice, this is exactly what I was looking for! I'd like to it from the change log though to make easier to discover.

I finished writing testing-docs now.  They are available at #64. I would link to them from the change log as well, as we improved our test-utils too.

File: CHANGELOG.md [hunk 4]
```diff
@@ -1,3 +1,124 @@
+## Draft for 2.0.0
+
+## BREAKING CHANGES
+
+This release introduce several breaking changes and requires to migrate some parts.
```
Comment (montezume) [inline CHANGELOG.md]: nitpick: `introduces` and `requires migration in some parts`

File: CHANGELOG.md [hunk 5]
```diff
@@ -1,3 +1,124 @@
...
+
+### Dropped some connectors and introduce a new connector for `ApplicationState`
```
Comment (montezume) [inline CHANGELOG.md]: nitpick: Since dropped is past tense, so should introduce be introduced

File: CHANGELOG.md [hunk 6]
```diff
@@ -1,3 +1,124 @@
...
+
+The `@commercetools-frontend/application-shell-connectors` used to contain some connectors that have now been **removed**:
```
Comment (montezume) [inline CHANGELOG.md]: nitpick: The package ...

File: CHANGELOG.md [hunk 7]
```diff
@@ -1,3 +1,124 @@
...
+
+- **configuration**: `injectConfiguration` and `<ConfigurationConsumer>`
+- **project-data-locale**: `withProjectDataLocale` and `<GetProjectDataLocale>`
+- **user-permissions**: `withUserPermissions` and `<GetUserPermissions>`
+- **user-time-zone**: `withUserTimeZone` and `<GetUserTimeZone>`
+
+As a replacement, we now have a single component `<GetApplicationState>` and `withApplicationState` that provide all the necessary "global" information about the `user`, `project` and application `environment`.
```
Comment (montezume) [inline CHANGELOG.md]: maybe reword this to say that we have a single component, and a single hoc? It's sort of confusing to say we have a single component and then list two.

Comment (montezume) [reply CHANGELOG.md]: nitpick: `provide` -> provides

File: CHANGELOG.md [hunk 8]
```diff
@@ -1,3 +1,124 @@
...
+#### `project` fields
+
+- `key`
+- `version`
+- `name`
+- `countries`
+- `currencies`
+- `languages`
+- `permissions`: an object containing boolean flags about the permissions of the logged in user for the selected project (e.g. `{ canViewProducts: true, canManageOrders: false, ... }`)
+- `dataLocale`: the selected project **locale** (from the locale switcher in the AppBar) used to render a localized field of the project data. The available values are based on the `project.languages`
```
Comment (montezume) [inline CHANGELOG.md]: 💯

File: CHANGELOG.md [hunk 9]
```diff
@@ -1,3 +1,124 @@
...
+
+The following are common fields defined in `env.json`. However, each application can provide more specific fields that cannot be documented.
+
+- `frontendHost`: the host where the Merchant Center application is running (e.g. `mc.commercetools.com`)
+- `mcApiUrl`: the API URL of the Merchant Center (`https://mc-api.commercetools.com` for projects in `EU` and `https://mc-api.commercetools.co` for projects in `US`)
+- `location`: the location where the Merchant Center is running, usually `eu` or `us`
+- `env`: the environment where the Merchant Center is running, usually `production` or `staging`
+- `cdnUrl`: the URL where the static assets are stored
+- `servedByProxy`: a flag to indicate if this application is running behind the Merchant Center proxy or not, usually `true` for production and `false` for local development
+
```
Comment (montezume) [inline CHANGELOG.md]: This is well done 👍

File: CHANGELOG.md [hunk 10]
```diff
@@ -1,3 +1,124 @@
...
+```js
+withApplicationState(applicationState => ({
+  projectKey: applicationState.project && applicationState.project.key,
+  userEmail: applicationState.user && applicationState.user.email,
+}))(MyComponent);
+```
+
+### Imports
+
+All packages now will expose **named exports** and do not allow to import from paths reaching inside the packages.
```
Comment (montezume) [inline CHANGELOG.md]: nitpick: will now

File: CHANGELOG.md [hunk 11]
```diff
@@ -1,3 +1,124 @@
...
+
+// After
+import { AsyncLocaleData } from '@commercetools-frontend/i18n';
+```
+
+> This includes also the `<ApplicationShell>`, so now it's `import { ApplicationShell } from '@commercetools-frontend/application-shell';`
+
+### Published packages are bundled to ESM or CJS
+
+Previously we were shipping untranspiled code, requiring consumers of the packages to instruct webpack to include those packages in the transpilation process.
```
Comment (montezume) [inline CHANGELOG.md]: 👍

File: CHANGELOG.md [hunk 12]
```diff
@@ -1,3 +1,124 @@
...
+];
+
+createWebpackConfigForDevelopment({
+  distPath,
+  entryPoint,
+  sourceFolders,
+});
+```
+
+Now that we ship transpiled code, it's not necessary anymore to include those packages.
```
Comment (montezume) [inline CHANGELOG.md]: include these packages ... in your webpack config?

**Comment by emmenko** [comment]
Thanks for the feedback guys, however I encourage you to just make the changes you want and push them directly to this PR (especially the nitpicks) 😉 Thanks!

File: CHANGELOG.md [hunk 13]
```diff
@@ -73,6 +73,10 @@ withApplicationState(applicationState => ({
 }))(MyComponent);
 ```
 
+### Renamed prop in `<ApplicationShell>`
```
Comment (montezume) [inline CHANGELOG.md]: 💯

**Comment by tdeekens** [comment]
> Now you know how @tdeekens feels with the English nitpicks though 😆

He's into it from tip to toe. Today he had a commit message `chore: fix english`. The man can single handedly fix the language!

Tomorrow, I am gonna fix the internet! https://www.youtube.com/watch?v=5kUNIDVIjh0

**Comment by dferber90** [comment]
I reworded and restructured the CHANGELOG a bit. Hope this helps to clarify the changes made.
